sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -467,10 +494,148 @@ The `node` class provides the {crs}[common reference semantics].
+it with new values.
+
+The type of the underlying object a dynamic parameter represents is set at
+compile time using a template parameter. This underlying type can be an
+accessor, a pointer to a USM allocation, scalar passed by value, or a raw byte
+representation of the argument. The raw byte representation is intended to
+enable updating arguments set using
+link:../proposed/sycl_ext_oneapi_raw_kernel_arg.asciidoc[sycl_ext_oneapi_raw_kernel_arg].
+
+Dynamic parameters are registered with nodes in a modifiable graph, with each
+registration associating one or more node arguments to the dynamic parameter
+instance. Registration happens inside the command-group that the node represents
+, and is done using the argument index (known when argument originally set using
+`set_arg()`/`set_args()`). It is valid for a node argument to be registered with
+more than one dynamic parameter instance.
+
+Registration is done by using the argument index (known when argument
+originally set using `set_arg()`/`set_args()`).
+
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+The `dynamic_parameter` class provides the {crs}[common reference semantics].
+
+Table {counter: tableNumber}. Member functions of the `dynamic_parameter` class.
+[cols="2a,a"]
+|===
+|Member Function|Description
+
+|
+[source,c++]
+----
+dynamic_parameter(command_graph<graph_state::modifiable> graph);
+----
+|Constructs a dynamic parameter object that can be registered with node
+arguments.
+
+Parameters:
+
+* `graph` - Graph containing the nodes which will have dynamic parameters.
+
+|
+[source,c++]
+----
+dynamic_parameter(const ValueT &initialValue, command_graph<graph_state::modifiable> graph);
+----
+|Constructs a dynamic parameter object that can be registered with node
+arguments, while providing an intiial value.
+
+Parameters:
+
+* `initialValue` - Initial value for the dynamic parameter.
+* `graph` - Graph containing the nodes which will have dynamic parameters.
+
+|
+[source,c++]
+----
+void register(int argIndex, const handler &cgh);
+----
+|Callable only from command-group scope when adding a node to a graph
+using the explicit API. Associate the dynamic parameter
+with the kernel parameter of the node that encapsulates this command-group.
+Parameters are associated by their index which was given when they were
+registered with `handler`.
+
+Parameters:
+
+* `argIndex` - Kernel argument index starting from zero to associate with
+  the dynamic parameter.
+* `handler` - SYCL handler object.
+
+Exceptions:
+
+* Throws with error code `invalid` if `argIndex` is not a valid argument index
+  for the resulting node.
+* Throws with error code `invalid` if the type of the command-group is not a
+  kernel execution.
+* Throws with error code `invalid` if the resulting node is not associated with
+  the graph that this dynamic parameter was created with.
+
+|
+[source,c++]
+----
+void update(const ValueT& newValue);
```

Comment:
There should be some way to update the kernel argument value using `raw_kernel_arg`.  I think you cannot use this overload because it expects a `ValueT`.  Do we need a second overload taking `raw_kernel_arg`?

Reply:
Would this not be covered by using `raw_kernel_arg` from the start, aka `dynamic_parameter<raw_kernel_arg>`? We may have to do some runtime checking of the size of the arg since we wouldn't have static type checking, but that would be fine.

Reply:
Resolving this as discussion was continued on another comment, and the outcome was leaving it as is for now.